docs/Environment-Variables.md

```diff
@@ -0,0 +1,43 @@
+# Environment Variables
+
+In order to configure our deployment and tools we'll set up some
+environment variables to ensure consistency. If you are running these
+scripts through SimDem you can customize these values by copying and
+editing `env.json` into `env.local.json`.
+
+We'll need a unique name for our Resource Group in Azure, but when
+running in an automated mode it is useful to have a (mostly) unique
+name for your deployment and related resources. We'll use a timestamp.
+If the environmnt variable `MOODLE_RG_NAME` is not set we will
+create a new value using a timestamp:
+
+
+``` shell
+if [ -z "$MOODLE_RG_NAME" ]; then MOODLE_RG_NAME=moodle_$(date +%Y-%m-%d-%H); fi
```

Comment:
Hourly resolution may not be enough any more?

Reply:
In which use case? 

For testing we would pass in fixed names. For manual execution we can pass in names but I don't think we are likely to be deploying more than once per hour. I can certainly add seconds, but then that precludes the possibility of rerunning a failed template without passing in values (which was my intent).

Reply:
Well, this might be a minor thing anyway, but I was mainly thinking of the faster deployment time with NFS and without vnet gateway, which can be as low as 30 minutes. With Redis being made optional, I suppose it could be even as low as 15 minutes. Still, I guess very few would deploy more than once per hour. In that regard, with load testing, I thought of cases where we initiate multiple deployments of the templates with varying params and run loadtesting on them concurrently, but that may not be supported on SimDem anyway? I need to know more about SimDem, in the first place... :)

Reply:
Yes, the testing use case is covered in SimDem - we can pass environment variables in via a config file. For now treat these does as "manual" tutorial docs. I'll document the use of SimDem eventually (we are currently re-writing it from my initial hack to something maintainable, so things may change if I were to document it now). However, if you want to explore the hacky version http://github.com/azure/simdem (the rewrite is in a branch called SimDem2, but these scripts don't run 100% in it yet).

I think I'll leave it like this for now. If we see a problem we can change in the future.